core/Tracker/TableLogAction/Cache.php

```diff
@@ -0,0 +1,172 @@
+<?php
+/**
+ * Piwik - free/libre analytics platform
+ *
+ * @link http://piwik.org
+ * @license http://www.gnu.org/licenses/gpl-3.0.html GPL v3 or later
+ *
+ */
+
+namespace Piwik\Tracker\TableLogAction;
+
+use Piwik\Common;
+use Piwik\Config;
+use Piwik\Container\StaticContainer;
+use Psr\Log\LoggerInterface;
+
+class Cache
+{
+    /**
+     * @var bool
+     */
+    public $isEnabled;
+
+    /**
+     * @var int cache lifetime in seconds
+     */
+    protected $lifetime;
+
+    /**
+     * @var LoggerInterface
+     */
+    private $logger;
+
+    /**
+     * for tests
+     *
+     * @var int
+     */
+    static public $hits = 0;
+
+    public function __construct()
+    {
+        $this->isEnabled = (bool)Config::getInstance()->General['enable_segments_subquery_cache'];
+        $this->limitActionIds = Config::getInstance()->General['segments_subquery_cache_limit'];
+        $this->lifetime = Config::getInstance()->General['segments_subquery_cache_ttl'];
+        $this->logger = StaticContainer::get('Psr\Log\LoggerInterface');
```

Comment:
If a class has dependencies, it should be stored in DI. This class should be stored in DI.